src/multi-select/tp-multi-select.ts

```diff
@@ -118,38 +123,35 @@ export class TPMultiSelectElement extends HTMLElement {
 	protected updateFormFieldValue(): void {
 		// Get options.
 		const styledSelectedOptions: NodeListOf<TPMultiSelectOptionElement> | null = this.querySelectorAll( `tp-multi-select-option` );
-		const selectFieldOptions: NodeListOf<HTMLOptionElement> | null = this.querySelectorAll( 'select option' );
+		const selectField: HTMLSelectElement | null = this.querySelector( 'select' );
 
-		if ( ! styledSelectedOptions || ! selectFieldOptions ) {
+		if ( ! styledSelectedOptions || ! selectField ) {
 			return;
 		}
 
 		// Traverse options.
 		styledSelectedOptions.forEach( ( option: TPMultiSelectOptionElement ): void => {
-			// Get matching select field options.
-			const matchingSelectOptions: HTMLOptionElement[] = [ ...selectFieldOptions ].filter( ( selectOption: HTMLOptionElement ): boolean =>
-				selectOption.getAttribute( 'value' ) === option.getAttribute( 'value' ) );
-
-			if ( 0 === matchingSelectOptions.length ) {
-				return;
-			}
-
-			// Check whether to mark them as selected or not.
-			if ( 'yes' === option.getAttribute( 'selected' ) ) {
-				matchingSelectOptions.forEach( ( matchingSelectOption: HTMLOptionElement ): void => {
-					matchingSelectOption.selected = true;
-					matchingSelectOption.setAttribute( 'selected', 'selected' );
-				} );
-			} else {
-				matchingSelectOptions.forEach( ( matchingSelectOption: HTMLOptionElement ): void => {
-					matchingSelectOption.selected = false;
-					matchingSelectOption.removeAttribute( 'selected' );
-				} );
+			const optionValue = option.getAttribute( 'value' ) ?? '';
+			if ( optionValue ) {
+				const matchingSelectOption: HTMLOptionElement | null = this.querySelector( `select option[value="${ optionValue }"]` );
```

Comment:
@i-am-chitti This is a lot of querySelectors which could lead to a fall in performance, depending on how many options there are - can we optimise this?

Reply:
Have tried keeping the query selectors minimum as it was before.